### PR TITLE
feat: add FASTA sequence download

### LINKED ISSUE #102: Provide sequence for sequence ID
FASTA format

### PR DESCRIPTION
**Pull request by jsstevenson**
close #102

### CODE REVIEW

File: uta_tools/main.py [hunk 1]
```diff
@@ -220,3 +225,36 @@ async def get_mapped_mane_data(
...
+    except KeyError:
+        raise HTTPException(
+            status_code=404,
+            detail="No sequence available for requested identifier"
+        )
+    background_tasks.add_task(
+        lambda p: os.unlink(p),
+        path
+    )
+    return FileResponse(path)
```
Comment (jsstevenson) [inline uta_tools/main.py:260]: This doesn't necessarily need to live in UTA Tools, but I wrote it to make sure that the file creation implementation would work with a FastAPI endpoint and left it in by default (I'll have to copy/paste to use in the curation page).

File: uta_tools/main.py [hunk 2]
```diff
@@ -220,3 +225,36 @@ async def get_mapped_mane_data(
...
+    f"/{SERVICE_NAME}/download_sequence",
+    summary="Get sequence for ID",
+    response_description=RESP_DESCR,
+    description="Given a known accession identifier, retrieve sequence data and return"
+                "as a FASTA file",
+    response_class=FileResponse
+)
+async def get_sequence(
+    background_tasks: BackgroundTasks,
+    sequence_id: str = Query(..., description="ID of sequence to retrieve"),
```
Comment (korikuzma) [inline uta_tools/main.py:240]: Might be useful to include the part about not including namespace

File: tests/unit/test_uta_tools.py [hunk 3]
```diff
@@ -768,3 +768,49 @@ async def test_invalid(test_uta_tools):
     assert resp.warnings == ["NM_182752.4 with position 2138 does not exist"
                              " on exons: [(58, 259), (259, 351), (351, 528),"
                              " (528, 682), (682, 877)]"]
+
+
+def test_get_fasta_file(test_uta_tools, tmp_path):
```
Comment (korikuzma) [inline tests/unit/test_uta_tools.py:773]: Should we add a test where the header is `gnl|ID|{sequence_id}`

**Review by korikuzma — APPROVED** [review]
+1

File: uta_tools/uta_tools.py [hunk 4]
```diff
@@ -569,3 +570,77 @@ def _get_exon_number(tx_exons: List, tx_pos: int) -> int:
...
+        """
+        other_ids, warnings = self.seqrepo_access.translate_identifier(sequence_id)
+        if warnings:
+            raise KeyError(f"Unable to find other IDs for {sequence_id}")
+        aliases = []
+        for prefix in prefixes:
+            for other_id in other_ids:
+                if other_id and other_id.startswith(prefix):
+                    aliases.append(other_id)
+        return aliases
```
Comment (korikuzma) [inline uta_tools/uta_tools.py:589]: I believe this does the same thing
```suggestion
        aliases, warnings = self.seqrepo_access.translate_identifier(sequence_id,
                                                                     prefixes)
        if warnings:
            raise KeyError(f"Unable to find other IDs for {sequence_id}")
        return aliases
```

File: uta_tools/uta_tools.py [hunk 5]
```diff
@@ -569,3 +570,77 @@ def _get_exon_number(tx_exons: List, tx_pos: int) -> int:
...
+        ]
+
+        if sequence_id[:3] in REFSEQ_PREFIXES:
+            aliases = self._get_fasta_aliases(sequence_id, ["ensembl", "ga4gh"])
+            header = f">ref|refseq:{sequence_id}|{'|'.join(aliases)}"
+        elif sequence_id[:4] in ENSEMBL_PREFIXES:
+            aliases = self._get_fasta_aliases(sequence_id, ["refseq", "ga4gh"])
+            header = f">emb|ensembl:{sequence_id}|{'|'.join(aliases)}"
+        else:
+            aliases = self._get_fasta_aliases(
```
Comment (korikuzma) [inline uta_tools/uta_tools.py:634]: If we wanted to add a test for this, we could use the seguid

**Review by korikuzma — DISMISSED** [review]

**Comment by jsstevenson** [comment]
alright! maybe finally done?

**Review by korikuzma — APPROVED** [review]
🚀

**Pull request merged**